### PR TITLE
Feat/add matches

### PR DESCRIPTION
**Pull request by O-tega**

### CODE REVIEW

File: app/Controllers/Http/MatchesController.ts [hunk 1]
```diff
@@ -1,3 +1,42 @@
...
+        try{
+
+            const user: User | null = auth.user ?? null
+
+            if(!user){
+                Logger.error({err:new Error('Not found')}, 'user is not found')
+                return notFoundResponse({response, message:'User is not found'})
+            }
+
+            const petId = uuidv4
```
Comment (Okeibunor) [inline app/Controllers/Http/MatchesController.ts:25]: should be 
```suggestion
            const petId = uuidv4()
```

File: app/Controllers/Http/MatchesController.ts [hunk 2]
```diff
@@ -1,3 +1,42 @@
...
+
+        try{
+
+            const user: User | null = auth.user ?? null
+
+            if(!user){
+                Logger.error({err:new Error('Not found')}, 'user is not found')
+                return notFoundResponse({response, message:'User is not found'})
+            }
+
```
Comment (Okeibunor) [inline app/Controllers/Http/MatchesController.ts:24]: add 

`const breeder_id = user.uuid`

File: app/Controllers/Http/MatchesController.ts [hunk 3]
```diff
@@ -1,3 +1,42 @@
...
+import User from 'App/Models/User'
+import Pet from 'App/Models/Pet'
+import Match from 'App/Models/Match'
 
-export default class MatchesController {}
+export default class MatchesController {
+
+    public async createMatch({auth, response, request}:HttpContextContract){
+
+        const{pet_id, breeder_id, accepted} = await request.validate(CreateMatchValidator)
```
Comment (Okeibunor) [inline app/Controllers/Http/MatchesController.ts:14]: 1. No need for `accepted` in the validator. It already has a default of false in the db
2. no need for breeder_id in the request validator too. The breeder id would come from the auth.user . I've shown how to do it in the comment below
3.  Only the pet_id is required in the validator

File: app/Controllers/Http/MatchesController.ts [hunk 4]
```diff
@@ -1,3 +1,42 @@
...
+                return notFoundResponse({response, message:'User is not found'})
+            }
+
+            const petId = uuidv4
+			const match = new Match()
+            match.uuid = petId
+            match.breeder_id = breeder_id
+            match.accepted = accepted
+
+
```
Comment (Okeibunor) [inline app/Controllers/Http/MatchesController.ts:31]: to save your changes, use 

`await match.save()`

File: app/Validators/CreateMatchValidator.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { schema, CustomMessages } from '@ioc:Adonis/Core/Validator'
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+
+export default class CreateMatchValidator {
+  constructor(protected ctx: HttpContextContract) {}
+
+  /*
```
Comment (Okeibunor) [inline app/Validators/CreateMatchValidator.ts:7]: remove all these comments

File: app/Validators/CreateMatchValidator.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+
+	pet_id: schema.string(),
+  breeder_id: schema.string(),
+  accepted: schema.boolean(),
+
+  })
+
+
+  /**
```
Comment (Okeibunor) [inline app/Validators/CreateMatchValidator.ts:36]: remove these multiline comments and unnecessary spaces too.

File: database/migrations/1645988437194_users.ts
Comment (Okeibunor) [inline database/migrations/1645988437194_users.ts:1]: so many changes in this file but nothing actually changed.

File: config/auth.ts [hunk 7]
```diff
@@ -100,7 +100,7 @@ const authConfig: AuthConfig = {
         | that time.
         |
         */
-        model: () => import('App/Models/user'),
+        model: () => import('App/Models/User'),
```
Comment (Okeibunor) [inline config/auth.ts:103]: why these changes though?

File: contracts/auth.ts [hunk 8]
```diff
@@ -5,7 +5,7 @@
  * file.
  */
 
-import user from 'App/Models/user'
+import user from 'App/Models/User'
```
Comment (Okeibunor) [inline contracts/auth.ts:8]: why this change as well?

File: app/Controllers/Http/MatchesController.ts [hunk 9]
```diff
@@ -1,3 +1,38 @@
...
+		try {
+			const user: User | null = auth.user ?? null
+
+			if (!user) {
+				Logger.error({ err: new Error('Not found') }, 'user is not found')
+				return notFoundResponse({ response, message: 'User is not found' })
+			}
+
+			const petId = pet_id
```
Comment (Okeibunor) [inline app/Controllers/Http/MatchesController.ts:20]: notice that I corrected this to be equal to the `pet_id` passed in the request body

File: contracts/auth.ts (hunk 8 above)
Comment (O-tega) [reply contracts/auth.ts:8]: Its was because the UserController was importing it as User and it wasnt recognisng it so I had to change it

**Review by Okeibunor — APPROVED** [review]
